server/protocols/jmap/src/main/java/org/apache/james/jmap/send/PostDequeueDecorator.java

```diff
@@ -77,13 +79,14 @@ public Mail getMail() {
     public void done(boolean success) throws MailQueueException {
         mailQueueItem.done(success);
         if (success && mandatoryJmapMetaDataIsPresent()) {
-            MessageId messageId = messageIdFactory.fromString((String) getMail().getAttribute(MailMetadata.MAIL_METADATA_MESSAGE_ID_ATTRIBUTE));
-            String username = (String) getMail().getAttribute(MailMetadata.MAIL_METADATA_USERNAME_ATTRIBUTE);
-            if (getMail().getAttribute(IS_DELIVERED) == null) {
+            Optional<?> optionalRawMessageId = retrieveMessageId();
+            MessageId messageId = messageIdFactory.fromString((String) optionalRawMessageId.get());
```

Comment:
calling `Optional.get()` directly without any check may not be a proper way

Reply:
There is a check in `mandatoryJmapMetaDataIsPresent()`

Reply:
why don't you cast it to string before?

Reply:
(using the attribute utils)